packages/template-retail-react-app/app/pages/product-list/index.jsx

```diff
@@ -84,115 +89,193 @@ const REFINEMENT_DISALLOW_LIST = ['c_isNew']
  */
 const ProductList = (props) => {
     const {
-        searchQuery,
-        productSearchResult,
+        // eslint-disable-next-line react/prop-types
+        isLoading: _isLoading, // NOTE: Ignore `getProps` isLoading state.
         // eslint-disable-next-line react/prop-types
         staticContext,
-        location,
-        isLoading,
         ...rest
     } = props
-    const {total, sortingOptions} = productSearchResult || {}
-
     const {isOpen, onOpen, onClose} = useDisclosure()
-    const [sortOpen, setSortOpen] = useState(false)
     const {formatMessage} = useIntl()
     const navigate = useNavigation()
     const history = useHistory()
     const params = useParams()
-    const {categories} = useCategories()
+    const location = useLocation()
     const toast = useToast()
     const einstein = useEinstein()
+    const {res} = useServerContext()
+    const wishlist = useWishlist()
+    const customerId = useCustomerId()
+    const [searchParams, {stringify: stringifySearchParams}] = useSearchParams()
+
+    /**************** Page State ****************/
+    const [filtersLoading, setFiltersLoading] = useState(false)
+    const [wishlistLoading, setWishlistLoading] = useState([])
+    const [sortOpen, setSortOpen] = useState(false)
+
+    const urlParams = new URLSearchParams(location.search)
+    let searchQuery = urlParams.get('q')
+    const isSearch = !!searchQuery
 
-    // Get the current category from global state.
-    let category = undefined
-    if (!searchQuery) {
-        category = categories[params.categoryId]
+    if (params.categoryId) {
+        searchParams._refine.push(`cgid=${params.categoryId}`)
+    }
+
+    /**************** Mutation Actions ****************/
+    const {mutate: createCustomerProductListItem} = useShopperCustomersMutation({
+        action: 'createCustomerProductListItem'
+    })
+    const {mutate: deleteCustomerProductListItem} = useShopperCustomersMutation({
+        action: 'deleteCustomerProductListItem'
+    })
+
+    /**************** Query Actions ****************/
+    const {isLoading, isFetching, data: productSearchResult} = useProductSearch(
```

Comment:
Can we make this loading more contextual since there are other loading state in the page? `isProductSearchLoading`.

Reply:
same with isFetching